esmvalcore/cmor/_fixes/icon/icon.py

```diff
@@ -0,0 +1,163 @@
+        if cube.coords(lat_name) and cube.coords(lon_name):
+            self._fix_lat_lon(cube, lat_name, lon_name)
+
+        # Fix scalar coordinates
+        self._fix_scalar_coords(cube)
+
+        # Fix metadata of variable
+        self._fix_var_metadata(cube)
+
+        return iris.cube.CubeList([cube])
+
+    def _fix_scalar_coords(self, cube):
+        """Fix scalar coordinates."""
+        if "height2m" in self.vardef.dimensions:
+            add_scalar_height_coord(cube, 2.0)
+        if "height10m" in self.vardef.dimensions:
+            add_scalar_height_coord(cube, 10.0)
+        if "typesi" in self.vardef.dimensions:
+            add_scalar_typesi_coord(cube, 'sea_ice')
+
+    def _fix_var_metadata(self, cube):
+        """Fix metadata of variable."""
+        if self.vardef.standard_name == '':
+            cube.standard_name = None
+        else:
+            cube.standard_name = self.vardef.standard_name
+        cube.var_name = self.vardef.short_name
+        cube.long_name = self.vardef.long_name
+        if cube.units != self.vardef.units:
+            cube.convert_units(self.vardef.units)
+
+    @staticmethod
+    def _fix_height(cube, plev_points_cube, plev_bounds_cube):
+        """Fix height coordinate of cube."""
+        air_pressure_points = plev_points_cube.core_data()
+
+        # Get bounds from half levels and reshape array
+        air_pressure_bounds = plev_bounds_cube.core_data()
+        air_pressure_bounds = da.stack(
+            (air_pressure_bounds[:, :-1], air_pressure_bounds[:, 1:]), axis=-1)
+
+        # Setup air pressure coordinate with correct metadata and add to cube
+        air_pressure_coord = iris.coords.AuxCoord(
+            air_pressure_points,
+            bounds=air_pressure_bounds,
+            var_name='plev',
+            standard_name='air_pressure',
+            long_name='pressure',
+            units=plev_points_cube.units,
+            attributes={'positive': 'down'},
+        )
+        cube.add_aux_coord(air_pressure_coord, np.arange(cube.ndim))
+
+        # Add 'positive' attribute to height so iris can determine Z axis
+        # correctly
+        cube.coord('height').attributes['positive'] = 'down'
+
+        # Reverse height axis so that index 0 is surface level
+        cube = iris.util.reverse(cube, 'height')
+        return cube
+
+    @staticmethod
+    def _fix_lat_lon(cube, lat_name, lon_name):
+        """Fix latitude and longitude coordinates of cube."""
+        lat = cube.coord(lat_name)
+        lon = cube.coord(lon_name)
+
+        # Fix metadata
+        lat.var_name = "lat"
+        lon.var_name = "lon"
+        lat.standard_name = "latitude"
+        lon.standard_name = "longitude"
+        lat.long_name = "latitude"
+        lon.long_name = "longitude"
+        lat.convert_units('degrees')
+        lon.convert_units('degrees')
+        lat.units = "degrees_north"
+        lon.units = "degrees_east"
+
```

Comment:
I wonder about bounds. It's probably better to leave this to the future, but perhaps put a comment or otherwise a pin in it? Btw the mesh support is in [Iris (main) now](https://github.com/SciTools/iris/blob/main/docs/src/whatsnew/latest.rst). So maybe we can revisit that soonish.

Reply:
The bounds are set correctly already by iris, there is nothing that we need to change.

Reply:
Ah yes, got it.